backend/btrixcloud/orgs.py

```diff
@@ -434,6 +497,82 @@ async def inc_org_time_stats(self, oid, duration, is_exec_time=False):
             {"_id": oid}, {"$inc": {f"{key}.{yymm}": duration}}
         )
 
+        if not is_exec_time:
+            return
+
+        org = await self.get_org_by_id(oid)
+        monthly_base_quota_passed = await self.exec_mins_quota_reached(
+            oid, include_extra=False
+        )
+
+        # If we're still within our monthly base quota, nothing to do
+        if not monthly_base_quota_passed:
+            return
+
+        # If there's no extra or gifted time available, nothing to do
+        if not org.giftedExecSecondsAvailable and not org.extraExecSecondsAvailable:
+            return
+
+        # If we've surpassed monthly base quota, use gifted and extra exec minutes
+        # in that order if available, track their usage per month, and recalculate
+        # extraExecSecondsAvailable and giftedExecSecondsAvailable as needed
+        monthly_quota_mins = await self.get_org_exec_mins_monthly_quota(oid)
+        monthly_quota_secs = monthly_quota_mins * 60
+
+        current_monthly_exec_secs = await self.get_this_month_crawl_exec_seconds(oid)
+        secs_over_quota = current_monthly_exec_secs - monthly_quota_secs
+
+        gifted_secs_available = org.giftedExecSecondsAvailable
+        if gifted_secs_available:
+            if secs_over_quota <= gifted_secs_available:
+                return await self.orgs.find_one_and_update(
+                    {"_id": oid},
+                    {
+                        "$inc": {
+                            f"giftedExecSeconds.{yymm}": secs_over_quota,
+                            "giftedExecSecondsAvailable": -secs_over_quota,
+                            # Reset crawlExecSecs back to quota since we're
+                            # applying the excess to gifted
+                            f"crawlExecSeconds.{yymm}": -secs_over_quota,
+                        }
+                    },
+                )
+
+            # If seconds over quota is higher than gifted minutes available,
+            # use all of the gifted minutes and then move on to extra minutes
+            await self.orgs.find_one_and_update(
+                {"_id": oid},
+                {
+                    "$inc": {
+                        f"giftedExecSeconds.{yymm}": gifted_secs_available,
+                        "giftedExecSecondsAvailable": -gifted_secs_available,
```

Comment:
This should always be 0, right?

Reply:
Yup, just seemed easier to do the updates in one inc rather than splitting them into an inc and a set, but maybe a set to 0 would be clearer here

Reply:
yeah, would be clearer, though could also leave a comment perhaps?

Reply:
Changed to a `$set`